docs/container-set-template.md

```diff
@@ -0,0 +1,66 @@
+# Container Set Template
+
+![alpha](assets/alpha.svg)
+
+> v3.1 and after
+
+A container set templates is similar to a normal container or script template, but allows you to specify multiple
+containers to run within a single pod.
+
+Because you have multiple containers within a pod, they will be scheduled on the same host. You can use cheap and fast
+empty-dir volumes instead of persistent volume claims to share data between steps.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: container-group-template-
+spec:
+  entrypoint: main
+  templates:
+    - name: main
+      volumes:
+        - name: workpsace
+          emptyDir: { }
+      containerSet:
+        volumeMounts:
+          - mountPath: /workspace
+            name: workspace
+        graph:
+          containers:
+            - name: a
+              image: argoproj/argosay:v2
+            - name: b
+              image: argoproj/argosay:v2
+            - name: main
+              image: argoproj/argosay:v2
+              dependencies:
+                - a
+                - b
```

Comment:
@simster7 reworked dag to use the more modern 

```
depends: A && B
```

syntax. I think we should use that instead to (1) be consistent (2) allow future improvement (e.g. `depends: A || B`)

For MVP, we can only support `&&` which is what the list would provide

Reply:
I'd like to do this as v1.1 feature.

Reply:
I want to avoid the situation where there are two syntaxes to support (`depends:` and `dependencies:`). So while it's fine we don't support all of the advanced operators such as `||` in v1.0, I am suggesting that we only support `depends:` but with `&&` support

Reply:
I'm polling the community on this on. My guts say we should do both, but lets see what the poll says.

Reply:
If `depends` is a superset of the capabilities of `dependencies`, is there a reason you want to have both `dependencies` and `depends`?